installer/scripts/service_control

```diff
@@ -20,29 +23,75 @@ VAR_DIR=/var/opt/microsoft/omsagent
 ETC_DIR=/etc/opt/microsoft/omsagent
 BIN_DIR=/opt/microsoft/omsagent/bin
 
-WORKSPACE_REGEX='^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$'
+REGEX_UUID='[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}'
+WORKSPACE_REGEX="^$REGEX_UUID$"
+VALIDATION_SUPPLEMENT_MESSAGE=
+VALIDATION_SEVERITY=WARNING
 
 # Space seperated list of non oms workspaces
 NON_OMS_WS="scom"
 
-setup_variables()
-{
+# Begin Lower Level Utilities
+
+validate_file_has_line_pattern() {
+	local _fSpec=$1
+	local _pattA="$2"
+	local _boolOnly=$3
+	cat $_fSpec | grep -Eq "^$_pattA$"
```

Comment:
If the pattern is matched, this should not print anything to stdout; can you validate that nothing gets printed out on this line?

Reply:
We could have a unit test for that.  I may not understand the question. -q is a standard grep switch.

Reply:
Ah, I wasn't familiar with the grep -q switch. That's fine.

Reply:
I actually saw it first somewhere in this codebase.  It is useful under non-Bash because we need grep to do regular expressions.  In modern Bash there are regular expressions in the [[ ]] segments available.